### PR TITLE
SOC-1785

### PR DESCRIPTION
**Pull request by rafalkalinski**
support for ellipse of long posts on FireFox and IE

https://wikia-inc.atlassian.net/browse/SOC-1785

### CODE REVIEW

**Comment by rafalkalinski** [comment]
I'm not sure about that... 
Share your thoughts.

**Comment by rogatty** [comment]
You can use [truncate helper](https://github.com/Wikia/mercury/blob/dev/front/scripts/main/helpers/truncate.js) for that.

File: front/scripts/mercury/utils/truncate.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+/**
+ * @param {Array} params
+ * @returns {string}
+ */
+export default function (params) {
```
Comment (kvas-damian) [inline front/scripts/mercury/utils/truncate.js]: Please split ```params``` into variables.

File: front/scripts/mercury/utils/truncate.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+		Ember.Logger.error(`Truncate Util expected string as a parameter, but ${typeof text} given:`, text);
+		return null;
+	}
+
+	if (text.length <= maxLength) {
+		return text;
+	}
+
+	truncatedString = text.substr(0, maxLength);
+	lastSpacePosition = truncatedString.lastIndexOf(' ');
```
Comment (kvas-damian) [inline front/scripts/mercury/utils/truncate.js]: What do you think about adding support for ```\n```?

File: front/scripts/mercury/utils/truncate.js (hunk 1 above)
Comment (rogatty) [inline front/scripts/mercury/utils/truncate.js]: Yup, this piece of code should be in main/helpers/truncate:
```
text = params[0],
maxLength = params[1]
```

File: front/scripts/mercury/utils/truncate.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+export default function (params) {
+	const text = params[0],
+		maxLength = params[1] || 48,
+		ellipsisCharacter = '\u2026';
+
+	let truncatedString,
+		lastSpacePosition;
+
+	if (typeof text !== 'string') {
+		Ember.Logger.error(`Truncate Util expected string as a parameter, but ${typeof text} given:`, text);
```
Comment (rogatty) [inline front/scripts/mercury/utils/truncate.js]: If you want this file to be in `front/scripts/mercury/utils` you can't use `Ember`. IMO it should live in `front/scripts/main/utils` as there is currently no reason to share it with auth pages.

File: front/scripts/mercury/utils/truncate.js (hunk 2 above)
Comment (rafalkalinski) [inline front/scripts/mercury/utils/truncate.js]: How about simple trim instead of checking if there is any whitespace at the end?

Comment (kvas-damian) [inline front/scripts/mercury/utils/truncate.js]: It's not just checking if whitespace is at the end. It looks for last full word and truncates there.

Comment (rafalkalinski) [inline front/scripts/mercury/utils/truncate.js]: Right!
Anything against checking of any whitespace (\s) there?

Comment (kvas-damian) [inline front/scripts/mercury/utils/truncate.js]: Should be ok

**Comment by kvas-damian** [comment]
What do you think about adding unit test for ```\n``` support?

File: front/scripts/main/mixins/discussion-parsed-content.js [hunk 4]
```diff
@@ -1,9 +1,18 @@
...
  * Handles sending upvote action outside from the component.
  */
 export default Ember.Mixin.create({
+
+	/**
+	 * Property used to truncate the post body to 148 chars.
+	 * This property is set only in Firefox and in IE, because in other browsers works 'line-clamp' css property.
+	 * This is hack for the browsers that do not support 'line-clamp'.
+	 */
+	isContentTruncationHack: (/Firefox|Trident|Edge/).test(navigator.userAgent),
```
Comment (bkoval) [inline front/scripts/main/mixins/discussion-parsed-content.js]: This name is a bit unclear to me - how about shouldUseTruncationHack?

**Comment by bkoval** [comment]
LGTM :+1:

**Comment by kvas-damian** [comment]
LGTM

**Pull request merged**